httpclient_android/http_client/src/main/java/com/backblaze/b2/client/webApiHttpClient/android/UrlConnection/B2WebApiHttpClientImpl.java

```diff
@@ -0,0 +1,414 @@
+     */
+    @Override
+    public B2Headers head(String url, B2Headers headersOrNull)
+            throws B2Exception {
+        try {
+            final URL clientUrl = new URL(url);
+            URLConnection head;
+            if (clientUrl.getProtocol().equals("https")) {
+                 head = (HttpsURLConnection) clientUrl.openConnection();
+            } else {
+                head = (HttpURLConnection) clientUrl.openConnection();
+            }
+            if (headersOrNull != null) {
+                makeHeaders(headersOrNull).forEach((name, value) -> head.setRequestProperty(name, value));
+            }
+
+            head.setDoOutput(true);
+
+            int statusCode = head.getResponseCode();
+            if (statusCode >= 200 && statusCode < 300) {
+                final B2HeadersImpl.Builder builder = B2HeadersImpl.builder();
+                Arrays.stream(head.getHeaderFields().entrySet().forEach(header -> builder.set(header.getName(), header.getValue())));
+                return builder.build();
+            } else {
+                throw B2Exception.create(null, statusCode, null, "");
+            }
+        } catch (IOException e) {
+            throw translateToB2Exception(e, url);
+        }
+        finally {
+            closeQuietly(response);
+        }
+    }
+
+    @Override
+    public void close() {
+        clientFactory.close();
+    }
+
+    private String postJsonAndReturnString(String url,
+                                           B2Headers headersOrNull,
+                                           Object request) throws B2Exception {
+        String requestString = parseToStringUsingBzJson(request);
+        return postAndReturnString(url, headersOrNull, requestString);
+    }
+
+    /**
+     * POSTs to a web service that returns content, and returns the content
+     * as a single string.
+     *
+     * @param url the url to post to
+     * @param headersOrNull the headers, if any.
+     * @param requestString the string to post.
+     * @return the body of the response.
+     * @throws B2Exception if there's any trouble
+     */
+    private String postAndReturnString(String url, B2Headers headersOrNull, String requestString)
+            throws B2Exception {
+
+        try {
+            URLConnection post = new URL(url).openConnection();
+            if (headersOrNull != null) {
+                makeHeaders(headersOrNull).forEach((name, value) -> post.setRequestProperty(name, value));
+            }
+            if (requestString != null) {
+                // Sets the URLConnection client request to send
+                // https://docs.oracle.com/javase/8/docs/api/java/net/URLConnection.html#setDoInput-boolean-
+                post.setDoOutput(true);
+                try(OutputStream os = post.getOutputStream()) {
+                    os.write(requestString, 0, requestString.length);
+                } catch (Exception e) {
+                    //
```

Comment:
swallowing the exception doesn't seem like the right thing to do. Elsewhere we throw a B2Exception.